src/NRedisStack/Auxiliary.cs

```diff
@@ -2,133 +2,128 @@
-                                            string? LibraryName)
-        {
-            var _db = redis.GetDatabase();
-            if (LibraryName == null) // the user wants to disable the library name and version sending
-                _setInfo = false;
+        return args.ToArray();
+    }
 
-            else // the user set his own the library name
-                _libraryName = $"NRedisStack({LibraryName};.NET_v{Environment.Version})";
+    // TODO: add all the signatures of GetDatabase
+    public static IDatabase GetDatabase(this ConnectionMultiplexer redis,
+        string? LibraryName)
+    {
+        var _db = redis.GetDatabase();
+        if (LibraryName == null) // the user wants to disable the library name and version sending
+            _setInfo = false;
 
-            return _db;
-        }
+        else // the user set his own the library name
+            _libraryName = $"NRedisStack({LibraryName};.NET_v{Environment.Version})";
 
-        private static void SetInfoInPipeline(this IDatabase db)
-        {
-            if (_libraryName == null) return;
-            Pipeline pipeline = new Pipeline(db);
-            _ = pipeline.Db.ClientSetInfoAsync(SetInfoAttr.LibraryName, _libraryName!);
-            _ = pipeline.Db.ClientSetInfoAsync(SetInfoAttr.LibraryVersion, GetNRedisStackVersion());
-            pipeline.Execute();
-        }
+        return _db;
+    }
 
-        public static RedisResult Execute(this IDatabase db, SerializedCommand command)
-        {
-            if (_setInfo)
-            {
-                _setInfo = false;
-                db.SetInfoInPipeline();
-            }
-            return db.Execute(command.Command, command.Args);
-        }
+    private static void SetInfoInPipeline(this IDatabase db)
+    {
+        if (_libraryName == null) return;
+        Pipeline pipeline = new Pipeline(db);
+        _ = pipeline.Db.ClientSetInfoAsync(SetInfoAttr.LibraryName, _libraryName!);
+        _ = pipeline.Db.ClientSetInfoAsync(SetInfoAttr.LibraryVersion, GetNRedisStackVersion());
+        pipeline.Execute();
+    }
 
-        public async static Task<RedisResult> ExecuteAsync(this IDatabaseAsync db, SerializedCommand command)
-        {
-            if (_setInfo)
-            {
-                _setInfo = false;
-                ((IDatabase)db).SetInfoInPipeline();
-            }
-            return await db.ExecuteAsync(command.Command, command.Args);
-        }
+    public static RedisResult Execute(this IDatabase db, SerializedCommand command)
+    {
+        if (!_setInfo) return db.Execute(command.Command, command.Args);
+        _setInfo = false;
+        db.SetInfoInPipeline();
+        return db.Execute(command.Command, command.Args);
+    }
+
+    public static async Task<RedisResult> ExecuteAsync(this IDatabaseAsync db, SerializedCommand command)
+    {
+        if (!_setInfo) return await db.ExecuteAsync(command.Command, command.Args);
```

Comment:
In this class I find the original logic easier to read. Now the `db.ExecuteAsync(...)` call is duplicated, so if we need to change it, it needs change in two places.

Reply:
It was a very good point. I fixed it. Thank you